packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,151 @@
+ * Licensed under the MIT License.
+ */
+
+import { SchemaFactory } from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+	export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+		node: [() => DeepTreeNode, schemaFactory.number],
+	}) {}
+	{
+		type _check = ValidateRecursiveSchema<typeof DeepTreeNode>;
+	}
+
+/**
+ * A wide tree node with all leaves having the same value
+ */
+export const WideTreeNode = schemaFactory.array(schemaFactory.number);
+export type WideTreeNode = NodeFromSchema<typeof WideTreeNode>;
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: undefined, value: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue), value: undefined });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	if (depth < 1) {
+		throw new Error("Depth must be greater than 0");
+	}
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode = tree;
+	let depth = 1;
+
+	while (currentNode.node !== undefined) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode.value ?? assert.fail("missing leaf value");
+	return { depth, value };
+}
+
+/**
+ * Update the deep tree with a new value
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
+	let currentNode = tree;
+
+	while (currentNode.node !== undefined) {
+		currentNode = currentNode.node;
+	}
+
+	currentNode.value = newValue;
+}
+
+/**
+ * Make a wide tree with a single layer
+ */
+function makeWideTree(width: number, leafValue: number): WideTreeNode {
+	const values = [];
+	for (let i = 0; i < width; i++) {
+		values.push(leafValue);
+	}
+	return new WideTreeNode({ values });
+}
+
+/**
+ * generate a wide tree with a single layer
+ */
+export function generateWideSimpleTree(width: number, leafValue: number): WideTreeNode {
+	return hydrate(WideTreeNode, makeWideTree(width, leafValue));
+}
+
+/**
+ * read the whole wide tree and return the width and the sum of all value in the leaf node
+ */
+export function readWideSimpleTree(tree: WideTreeNode): {
+	nodesCount: number;
+	sum: number;
+} {
+	const nodesCount = tree.values.length;
+	let sum = 0;
+	for (const value of tree.values) {
+		sum += value;
+	}
+	return { nodesCount, sum };
+}
+
+/**
+ * write the whole wide tree with a new value
+ */
+export function writeWideSimpleTree(tree: WideTreeNode, newValue: number): void {
```

Comment:
From the names of "writeWideSimpleTree" and "writeWideSimpleTreeNewValue" its hard to tell that one replaces all the children (one at a time as separate edits) and the other does just 2 edits.

I think the tests might be more clear if instead of calling these utilities (which need more specific names to be clear) you just inline the editing logic into the test so its easier to change and/or make new different tests, and also easier to see whats being tested if you are reading the test.


Also I think this function is unused: avoiding leaving unused helpers around is another benefit of not making simple helpers like this.

Reply:
Removed it from here. Thx!